web/src/components/shared/NavBar.jsx

```diff
@@ -40,12 +41,12 @@ export class NavBar extends PureComponent {
     const { pathname } = this.props.location;
     if (pathname !== lastProps.location.pathname) {
       this.setSelectedTab();
+      this.getKotsApp();
```

Comment:
we should try to find a solution other than calling these functions on every update.